src/services/logger.service.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { buildAppLogger, buildWalletLogger } from '../logger';
+
+/**
+ * Wallet loggers mapped by walletId
+ * @type {Map<string, import('winston').Logger>}
+ */
+const walletLoggers = new Map();
+
+/**
+ * Initialize a wallet logger
+ * @param {string} walletId
+ */
+function initializeWalletLogger(walletId) {
+  const logger = buildWalletLogger(walletId);
+  walletLoggers.set(walletId, logger);
+}
+
+/**
+ * @param {import('express').Request} req
+ * @returns {import('winston').Logger}
+ */
+function getLogger(req) {
+  let walletId;
+
+  if (req.walletId) {
+    walletId = req.walletId;
+  } else if (req.headers['x-wallet-id']) {
+    walletId = req.headers['x-wallet-id'];
+  } else if (req.body && req.body['wallet-id']) {
+    walletId = req.body['wallet-id'];
+  } else {
+    // There is no wallet for this request
+    // will return the app logger instance.
+    return buildAppLogger();
+  }
+
+  const logger = walletLoggers.get(walletId);
+  if (!logger) {
+    // Uninitialized wallet, should use the app logger.
+    return buildAppLogger();
```

Comment:
Can we have here the app logger with the walletId parameter? Maybe we could have a log with `non-initialized wallet {wallet-id}` or something like that in this case. What do you think?

Reply:
Done